### PR TITLE
Build `flaky-test-reporter` in the prow-tests image

### PR DESCRIPTION
**Pull request by upodroid**
/cc @kvmware 

I'll clean up the makefiles and dockerfiles in a separate PR after the jobs for it is gone.

### CODE REVIEW

File: prow/jobs/custom/infra.yaml [hunk 1]
```diff
@@ -303,46 +303,6 @@ periodics:
 
 postsubmits:
   knative/infra:
-  - name: post-knative-infra-image-push
```
Comment (kvmware) [inline prow/jobs/custom/infra.yaml:306]: If you are removing this, I am worried about the comment in the `push-all.sh` file which says that prow will crash out without it?

Comment (upodroid) [reply prow/jobs/custom/infra.yaml:306]: look at the first line of that script.

```
# prow-tests image crashes prow job, don't build it here
EXCLUDED="prow-tests"
```

Comment (upodroid) [reply prow/jobs/custom/infra.yaml:306]: That should read, prow-tests image is huge and needs a separate job for it

**Pull request merged**